### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by royceschultz**

### CODE REVIEW

**Comment by royceschultz** [comment]
This is a major refactor that will break any existing code. It breaks up a few files and renames some methods. I think it's generally easier to use, but let me know what you think. 
It would be nice to query by show name or by date. Show name should be fine here, but we'll have to add an index for date or use a database.
Also, not a great commit history. I should have taken snapshots at each step.

**Comment by royceschultz** [comment]
I want your subjective opinion on how the code should feel. What patterns will make this more scalable or more unit-testable?
I'll probably throw out this branch and re-do it to avoid a whole lot of merge conflicts with the main branch, and to make a better commit history.

**Comment by chacetrev10** [comment]
I think it would be better if we changed "scrape.py" to EpisodeScrape.py or something like that. The episode.py are a lot of helper functions to the scrape.py so maybe that change would show the correlation better. Other than that the content of each file makes sense as all functions are similar.

File: MediaSentiment/Scrape.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        'cursor': cursor,
+    }
+    res = requests.get(BASE_URL + '/services/search/v1/scrape', payload)
+    assert(res.status_code == 200)
+    return json.loads(res.text)
+
+
+def allEpisodes(max_len=100000):
+    '''
+    Uses searchSegments until max_len episodes are found or no cursor object is returned (meaning we have fetched all results).
```
Comment (AndrewYeeYee) [inline MediaSentiment/Scrape.py:63]: Update docstring to updated function name.

File: MediaSentiment/Datastore.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    num_failed = 0
+    for i, episode in enumerate(episodes):
+        try:
+            print(f'Fetching Episode {i} ({i/len(episodes):.2%} done, {num_failed/len(episodes):.2%} failed)', end='\r')
+            episode_data = Scrape.getEpisode(episode['identifier'])
+            show = episode_data['metadata']['Title']
+            datetime = episode_data['metadata']['AirDate']
+            # Format string for Windows file system.
+            datetime = datetime.replace(',','').replace(' ', '_').replace(':','')
+            dirname = f'{folder_name}/{show}/{datetime}.json'
```
Comment (AndrewYeeYee) [inline MediaSentiment/Datastore.py:38]: change dirname to filepath

File: MediaSentiment/Datastore.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import os
+import json
+import logging
+
+if not os.path.exists('logs'):
+    os.makedirs('logs')
+logging.basicConfig(filename='logs/Archive.log', level=logging.INFO, format='%(asctime)s:%(levelname)s:%(message)s')
+
+
+def save(text, dirname):
```
Comment (AndrewYeeYee) [inline MediaSentiment/Datastore.py:13]: Changing dirname to filepath makes a bit more sense

File: MediaSentiment/Scrape.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        'fields': 'date,forumSubject,title,identifier',
+        'sorts': 'date',
+        'cursor': cursor,
+    }
+    res = requests.get(BASE_URL + '/services/search/v1/scrape', payload)
+    assert(res.status_code == 200)
+    return json.loads(res.text)
+
+
+def allEpisodes(max_len=100000):
```
Comment (AndrewYeeYee) [inline MediaSentiment/Scrape.py:61]: Maybe searchAllEpisodes is a bit more consistent with the above function names.

File: MediaSentiment/Scrape.py (hunk 1 above)
Comment (AndrewYeeYee) [inline MediaSentiment/Scrape.py:63]: Update docstring to updated function name.

File: MediaSentiment/Datastore.py (hunk 2 above)
Comment (AndrewYeeYee) [inline MediaSentiment/Datastore.py:38]: change dirname to filepath

File: MediaSentiment/Datastore.py (hunk 3 above)
Comment (AndrewYeeYee) [inline MediaSentiment/Datastore.py:13]: Changing dirname to filepath makes a bit more sense.

**Comment by royceschultz** [comment]
Closing because these changes are outdated and are being addressed in #18.

**Pull request closed without merging**